pkg/image/source_test.go

```diff
@@ -102,3 +142,135 @@ func TestParseSource(t *testing.T) {
 		}
 	}
 }
+
+func getDummyTar(t *testing.T, fs *afero.MemMapFs, paths ...string) string {
+	archivePath := "/image.tar"
+	testFile, err := fs.Create(archivePath)
+	if err != nil {
+		t.Fatalf("failed to create dummy tar: %+v", err)
+	}
+
+	tarWriter := tar.NewWriter(testFile)
+	defer tarWriter.Close()
+
+	for _, filePath := range paths {
+		header := &tar.Header{
+			Name: filePath,
+			Size: 13,
+		}
+
+		err = tarWriter.WriteHeader(header)
+		if err != nil {
+			t.Fatalf("could not write dummy header: %+v", err)
+		}
+
+		_, err = io.Copy(tarWriter, strings.NewReader("hello, world!"))
+		if err != nil {
+			t.Fatalf("could not write dummy file: %+v", err)
+		}
+	}
+
+	return archivePath
+}
+
+func getDummyPath(t *testing.T, fs *afero.MemMapFs, paths ...string) string {
+	dirPath := "/image"
+	err := fs.Mkdir(dirPath, os.ModePerm)
+	if err != nil {
+		t.Fatalf("failed to create dummy tar: %+v", err)
+	}
+
+	for _, filePath := range paths {
+		f, err := fs.Create(path.Join(dirPath, filePath))
+		if err != nil {
+			t.Fatalf("unable to create file: %+v", err)
+		}
+
+		if _, err = f.WriteString("hello, world!"); err != nil {
+			t.Fatalf("unable to write file")
+		}
+
+		if err = f.Close(); err != nil {
+			t.Fatalf("unable to close file")
+		}
+	}
+
+	return dirPath
+}
+
+func TestDetectSourceFromPath(t *testing.T) {
+	tests := []struct {
+		name           string
+		paths          []string
+		expectedSource Source
+		sourceType     string
+	}{
```

Comment:
it doesn't seem like there is a test case for:

* a path that doesn't exist
* an error produced by `fs.Open()`, I imagine this could be a permissions one? not sure if is feasible
* error from `seek`

Reply:
for a) there are tests for paths that do not exist, for instance:
```golang
		{
			name:           "oci-layout tar path",
			paths:          []string{"oci-layout"},
			sourceType:     "tar",
			expectedSource: OciTarballSource,
		},
```
this test case will run through checking for a non-existent `manifest.json` file. The `dummy dir paths` and `dummy tar paths` tests will ensure it behaves correctly when all paths will miss.

I've added a test for a non-existent path and changed the behavior around it --given the changed logic I think this will cover the missing fs.Open test (b).

I haven't added a test for a Seek error (c), but feel that the level of testing is probably sufficient.